test/e2e/httpproxy/015_http_health_checks_test.go

```diff
@@ -64,11 +65,17 @@ func testHTTPHealthChecks(fx *e2e.Framework) {
 	require.Truef(t, ok, "expected 200 response code, got %d", res.StatusCode)
 
 	// set the health check policy to always fail
-	require.NoError(t, fx.Client.Get(context.TODO(), client.ObjectKeyFromObject(p), p))
-	p.Spec.Routes[0].HealthCheckPolicy = &contourv1.HTTPHealthCheckPolicy{
-		Path: "/status/418",
-	}
-	require.NoError(t, fx.Client.Update(context.TODO(), p))
+	require.NoError(t, retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := fx.Client.Get(context.TODO(), client.ObjectKeyFromObject(p), p); err != nil {
+			return err
+		}
+
+		p.Spec.Routes[0].HealthCheckPolicy = &contourv1.HTTPHealthCheckPolicy{
+			Path: "/status/418",
+		}
+
+		return fx.Client.Update(context.TODO(), p)
+	}))
```

Comment:
Since this logic is ~repeated (get object, apply changes, update object), we could extract this to a test framework helper method. I was undecided on whether it was worth it or not, since you'd have to use some type-assertions or closure tricks to make it generic for types beyond just `HTTPProxy`.

Reply:
The status updating code had a similar problem, and I used a Mutator function there to handle it, I'm not sure how good I feel about it, looking back. Although this is repetitive, this does have the advantage of clarity. I agree we should leave this for now, and see if we should dry it up later.